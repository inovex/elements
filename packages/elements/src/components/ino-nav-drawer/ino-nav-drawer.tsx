import { MDCDrawer } from '@material/drawer';
import {
  Component,
  ComponentInterface,
  Element,
  Host,
  Prop,
  Watch,
  Event,
  EventEmitter,
  h,
} from '@stencil/core';
import { NavDrawerAnchor, NavDrawerVariant } from '../types';
import classNames from 'classnames';

/**
 * @slot header - For a custom header on top of the navigation bar
 * @slot logo - For the logo on top of the navigation bar (cannot be used with the `header` slot)
 * @slot subtitle - For the element just below the logo (cannot be used with the `header` slot)
 * @slot content - For the content of the navigation bar (usually used with `ino-list` and `ino-nav-item`)
 * @slot footer - For elements below the content slot
 * @slot app - For the application located next to this nav-drawer
 */
@Component({
  tag: 'ino-nav-drawer',
  styleUrl: 'ino-nav-drawer.scss',
  shadow: true,
})
export class NavDrawer implements ComponentInterface {
  /**
   * An internal instance of the material design drawer.
   */
  private drawerInstance: MDCDrawer;
  private drawerEl: HTMLElement;

  @Element() el!: HTMLElement;

  /**
   * Marks this element as open. (**unmanaged**)
   */
  @Prop() open?: boolean = false;

  @Watch('open')
  openChanged(newOpen: boolean) {
    if (this.drawerInstance) {
      this.drawerInstance.open = newOpen;
    }
  }

  /**
   * Side from which the drawer will appear.
   * Possible values: `left` (default), `right`.
   */
  @Prop() anchor?: NavDrawerAnchor = 'left';

  /**
   * The variant to use for the drawer
   * Possible values: `docked` (default), `dismissible`, `modal`.
   */
  @Prop() variant?: NavDrawerVariant = 'docked';

  componentDidLoad() {
    this.drawerInstance = new MDCDrawer(
      this.el.shadowRoot.querySelector('.mdc-drawer')
    );
    // set initial value of open state
    if (this.drawerInstance) {
      this.drawerInstance.open = this.open || false;
    }

    this.drawerEl.addEventListener('MDCDrawer:closed', this.closeDrawer);
  }

  disconnectedCallback() {
    this.drawerEl.removeEventListener('MDCDrawer:closed', this.closeDrawer);
    this.drawerInstance?.destroy();
  }

  /**
   * Emits when the user clicks on the drawer toggle icon to change the open state. Contains the status in `event.detail`.
   */
  @Event() openChange!: EventEmitter<boolean>;

  closeDrawer = (e: Event) => {
    e.preventDefault();
    this.openChange.emit(false);
  };

  toggleDrawer = (e: Event) => {
    const newOpenState = !this.open;
    this.openChange.emit(newOpenState);
    e.stopPropagation();
  };

  render() {
    const { anchor, variant } = this;

    const classDrawer = classNames({
      'mdc-drawer': true,
      'mdc-drawer--docked': variant === 'docked',
      'mdc-drawer--dismissible':
        variant === 'dismissible' || variant === 'docked', // docked is a modifier of MDC's dismissible inoVariant
      'mdc-drawer--modal': variant === 'modal',
      'mdc-drawer--anchor-left': anchor === 'left',
      'mdc-drawer--anchor-right': anchor === 'right',
    });

    const classAppContent = classNames({
      'mdc-drawer-app-content':
        variant === 'docked' || variant === 'dismissible',
    });

    const nav = (
      <aside class={classDrawer} ref={(el) => (this.drawerEl = el)}>
        <div class="mdc-drawer__header">
          <slot name="header">
            <div class="mdc-drawer__logo">
              <slot name="logo"></slot>
            </div>
            <div class="mdc-drawer__subtitle">
              <slot name="subtitle"></slot>
            </div>
          </slot>
        </div>

        <nav class="mdc-drawer__content">
          <slot name="content"></slot>
        </nav>

        <div class="mdc-drawer__footer">
          <slot name="footer"></slot>
          <ino-icon-button
            class="mdc-drawer__toggle"
            icon="arrow_right"
            onClick={this.toggleDrawer}
          />
        </div>
      </aside>
    );

    const main = (
      <div class={classAppContent}>
        <slot name="app"></slot>
      </div>
    );

    return (
      <Host>
        {nav}
        {variant === 'modal' && <div class="mdc-drawer-scrim"></div>}
        {main}
      </Host>
    );
  }
}
