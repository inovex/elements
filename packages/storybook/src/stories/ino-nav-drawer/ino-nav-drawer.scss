.story-nav-drawer__default,
.story-nav-drawer__docked_anchor--right,
.story-nav-drawer__dismissible {
  position: relative;
  height: 700px;
  border: 1px solid grey;
  border-radius: 4px;
  overflow: hidden;

  .main-content {
    box-sizing: border-box;
    overflow: auto;
    height: 100%;
    padding: 30px;
  }
}

.button__label {
  pointer-events: none;
}
