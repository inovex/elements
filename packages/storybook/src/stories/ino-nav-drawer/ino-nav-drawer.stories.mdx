import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-nav-drawer/readme.md";

<Meta title="Structure/ino-nav-drawer/Documentation" />

<Description markdown={readme} />
