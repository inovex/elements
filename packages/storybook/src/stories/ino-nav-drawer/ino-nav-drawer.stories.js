import { boolean, select, text } from '@storybook/addon-knobs';

import componentReadme from '_local-elements/src/components/ino-nav-drawer/readme.md';
import withStencilReadme from '_local-storybookcore/with-stencil-readme';

import './ino-nav-drawer.scss';

// == event block
const openChangeHandle = function (e) {
  const el = e.target;
  if (el.tagName.toLowerCase() !== 'ino-nav-drawer') {
    return;
  }
  el.setAttribute('ino-open', !!e.detail);
};

const clickHandler = function (e) {
  e.preventDefault();
  e.stopPropagation();

  const el = e.target;
  if (el.className.includes('toggle-nav')) {
    // go up to the ino drawer
    const drawer = el.parentElement.parentElement;
    const oldState = drawer.getAttribute('ino-open');
    drawer.setAttribute('ino-open', oldState === 'true' ? 'false' : 'true');
    return;
  }
  // variant: modal, click on scrim
  if (el.className.includes('nav-drawer-modal')) {
    el.setAttribute(
      'ino-open',
      el.getAttribute('ino-open') === 'true' ? 'false' : 'true',
    );
    return;
  }

  if (el.tagName.toLowerCase() === 'ino-list-item') {
    const parentDrawer =
      el.parentElement.parentElement.parentElement.parentElement;
    const navItemNodes = parentDrawer.querySelectorAll('ino-list-item');
    navItemNodes &&
      navItemNodes.forEach((link) => {
        link.setAttribute('ino-activated', false);
      });
    el.setAttribute('ino-activated', true);
    return;
  }
};

function unsubscribeFromComponentEvents() {
  document.removeEventListener('openChange', openChangeHandle);
  document.removeEventListener('click', clickHandler);
  document.removeEventListener('clickEl', clickHandler);
}

function subscribeToComponentEvents() {
  document.addEventListener('openChange', openChangeHandle);
  document.addEventListener('click', clickHandler);
  document.addEventListener('clickEl', clickHandler);

  // unsubscribe function will be called by Storybook
  return unsubscribeFromComponentEvents;
}

export default {
  title: 'Structure/<ino-nav-drawer>',
  parameters: {
    actions: {
      handles: [
        'openChange .customizable-drawer',
        'click .toggle-nav',
        'clickEl ino-nav-item',
      ],
    },
  },
  decorators: [
    (story) => {
      subscribeToComponentEvents();
      return story();
    },
  ],
};

const navContent = /*html*/ `
<ino-img slot="logo"
    src="https://app-uat.james.inma.inovex.io/static/media/james-logo.e7e70758.png"
    width="50"
></ino-img>
<div slot="subtitle">inovex</div>

<ino-list slot="content">
  <ino-nav-item ino-text="Some Link" ino-activated>
    <ino-icon
      ino-icon="onboarding"
    ></ino-icon>
  </ino-nav-item>
  <ino-nav-item ino-text="Some other Link">
    <ino-icon
      ino-icon="calendar"
    ></ino-icon>
  </ino-nav-item>
  <ino-nav-item ino-text="Some other Link">
    <ino-icon
      ino-icon="employee"
    ></ino-icon>
  </ino-nav-item>
  <ino-list-divider></ino-list-divider>
  <ino-nav-item ino-text="Some settings link">
    <ino-icon
      ino-icon="settings"
    ></ino-icon>
  </ino-nav-item>
</ino-list>

<ino-list slot="footer">
  <ino-nav-item ino-text="My Profile">
    <ino-img src="https://picsum.photos/id/1027/250/250.jpg" style="border-radius: 50%" ino-ratio-width="1" ino-ratio-height="1"></ino-img>
  </ino-nav-item>
</ino-list>
`;

const mainContent = /* html */ `
<p>At some point someone will replace this block of text with useful words so customers can learn more about the products and services you offer! Placeholder text is useful when you need to see what a page design looks like, but the actual content isn't available. It's like having someone with identical measurements check the fit of a dress before trying it on yourself. </p> <p>You are currently reading Honest Ipsum, the placeholder text that needs no explanation. Determining whether the typeface works or not is only possible if there is text for it to be applied to. This is just temporary placeholder text; like when a friend saves a spot for you in line, only to be replaced by you when you return. </p> <p>A web developer will often use filler text so they can focus on the design of the web page. It will be replaced with real content later. It is useful for a web developer to use placeholder text so they can easily see what different fonts look like on a realistic paragraph. </p> <p>Once the final copy for the web page has been created, it will go here. You are currently reading some filler text. A web developer will often use filler text so they can focus on the design of the web page. It will be replaced with real content later. </p> <p>We aren't quite sure what to put here yet. Determining whether the typeface works or not is only possible if there is text for it to be applied to. This text is only here to validate the layout. It isn't worth reading. What you are reading now is not what you will be reading in this space once this web page is completed. </p> <p>We aren't quite sure what to put here yet. Determining whether the typeface works or not is only possible if there is text for it to be applied to. This text is only here to validate the layout. It isn't worth reading. What you are reading now is not what you will be reading in this space once this web page is completed. </p>
`;

export const DefaultUsage = () => /* html */ `
      <style>
        .customizable-drawer {
          --ino-nav-drawer-background: ${text(
            '--ino-nav-drawer-background',
            '#ffffff',
            'Custom Properties',
          )};
          --ino-nav-drawer-text-color: ${text(
            '--ino-nav-drawer-text-color',
            '#003c7e',
            'Custom Properties',
          )};
          --ino-nav-drawer-width-open: ${text(
            '--ino-nav-drawer-width-open',
            '250px',
            'Custom Properties',
          )};
          --ino-nav-drawer-width-closed: ${text(
            '--ino-nav-drawer-width-closed',
            '72px',
            'Custom Properties',
          )};
          --ino-nav-drawer-height: ${text(
            '--ino-nav-drawer-height',
            '100%',
            'Custom Properties',
          )};
          --ino-nav-drawer-transition-duration: ${text(
            '--ino-nav-drawer-transition-duration',
            '0.25s',
            'Custom Properties',
          )};
          --ino-nav-drawer-timing-function: ${text(
            '--ino-nav-drawer-timing-function',
            '(0.4, 0, 0.2, 1)',
            'Custom Properties',
          )};
        }

        ino-nav-item {
          --ino-nav-item-color: ${text(
            '--ino-nav-item-color',
            'gray',
            'Custom Properties',
          )};
          --ino-nav-item-color-active: ${text(
            '--ino-nav-item-color-active',
            '#3d40f5',
            'Custom Properties',
          )};
        }
      </style>

      <div class="story-nav-drawer">
        <h4>Customizable</h4>
        <div class="story-nav-drawer__default">
          <ino-nav-drawer
            ino-open=${boolean('ino-open', false, 'Props')}
            ino-anchor="${select(
              'ino-anchor',
              { left: 'left', right: 'right' },
              'left',
              'Props',
            )}"
            ino-variant="${select(
              'ino-variant',
              { docked: 'docked', dismissible: 'dismissible', modal: 'modal' },
              'docked',
              'Props',
            )}"
            class="customizable-drawer"
          >
            <div slot="header">
              <ino-img
                src="${text(
                  'header - image src',
                  'https://app-uat.james.inma.inovex.io/static/media/james-logo.e7e70758.png',
                  'Slots',
                )}"
                width="50"
              ></ino-img>
              <p>inovex</p>
            </div>
            <ino-list slot="content">
              <ino-nav-item ino-text="${text(
                'default - 1st list item',
                'Some Link',
                'Slots',
              )}" ino-activated>
                <ino-icon
                  ino-icon="onboarding"
                ></ino-icon>
              </ino-nav-item>
            </ino-list>
            <ino-list slot="footer">
              <ino-nav-item ino-text="${text(
                'footer - 1st list item',
                'My Profile',
                'Slots',
              )}">
              <ino-icon ino-icon="employee"></ino-icon>
              </ino-nav-item>
            </ino-list>

            <main slot="app" class="main-content">
              <ino-button ino-fill="outline" ino-dense class="toggle-nav">Toggle Navigation</ino-button>
              <br /> <br />
              ${text('App Content', `Your App goes here ðŸ¤˜`, 'Slots')}
              ${mainContent}
              <br/><br />
            </main>
          </ino-nav-drawer>
        </div>

        <h4>Variant: Docked, right side</h4>
        <div class="story-nav-drawer__docked_anchor--right">
          <ino-nav-drawer
            ino-open="false"
            ino-anchor="right"
            class="nav-drawer-docked-right"
          >
            ${navContent}
            <main slot="app" class="main-content">
              ${mainContent}
            </main>
          </ino-nav-drawer>
        </div>
      </div>
  `;

DefaultUsage.decorators = [withStencilReadme(componentReadme)];

export const Dismissible = () => {
  return /* html */ `
    <h4>Variant: Dismissble</h4>
    <div class="story-nav-drawer__dismissible">
      <ino-nav-drawer
        ino-open="false"
        ino-variant="dismissible"
        class="nav-drawer-dismissible"
      >
       ${navContent}
       <main slot="app" class="main-content">
         <ino-button ino-fill="outline" ino-dense class="toggle-nav">Toggle Navigation</ino-button>
         ${mainContent}
       </main>
      </ino-nav-drawer>
    </div>
  `;
};

export const Modal = () => {
  return /* html */ `
      <div class="story-nav-drawer__modal">
        <ino-nav-drawer
          ino-open="false"
          ino-variant="modal"
          class="nav-drawer-modal"
        >
        ${navContent}
        <main slot="app" class="main-content">
          <h4>Variant: Modal</h4>
          <ino-button ino-fill="outline" ino-dense class="toggle-nav">Toggle Navigation</ino-button>
           ${mainContent}
         </main>
        </ino-nav-drawer>
      </div>
    `;
};
