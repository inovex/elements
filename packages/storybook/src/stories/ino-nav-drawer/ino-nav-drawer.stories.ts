import { useEffect } from '@storybook/client-api';
import { html } from 'lit-html';
import './ino-nav-drawer.scss';

const openChangeHandle = function (e) {
  const el = e.target;
  if (el.tagName.toLowerCase() !== 'ino-nav-drawer') {
    return;
  }
  el.setAttribute('open', !!e.detail);
};

const clickHandler = (e) => {
  e.preventDefault();
  e.stopPropagation();

  const el = e.target;
  if (el.className.includes('toggle-nav')) {
    // go up to the ino drawer
    const drawer = el.closest('ino-nav-drawer');
    drawer.open = !drawer.open;
    return;
  }

  // variant: modal, click on scrim
  if (el.className.includes('nav-drawer-modal')) {
    el.open = !el.open;
    return;
  }

  // select nav-drawer item
  if (el.tagName.toLowerCase() === 'ino-list-item') {
    const parentDrawer = el.closest('ino-nav-drawer');
    const navItemNodes = parentDrawer.querySelectorAll('ino-list-item');
    navItemNodes &&
      navItemNodes.forEach((link) => {
        link.activated = false;
      });
    el.activated = true;
    return;
  }
};

export default {
  title: `Structure/ino-nav-drawer`,
  component: 'ino-nav-drawer',
  parameters: {
    actions: {
      handles: [
        'openChange .customizable-drawer',
        'click .toggle-nav',
        'clickEl ino-nav-item',
      ],
    },
  },
  decorators: [
    (story) => {
      useEffect(() => {
        document.addEventListener('openChange', openChangeHandle);
        document.addEventListener('click', clickHandler);
        document.addEventListener('clickEl', clickHandler);

        return () => {
          document.removeEventListener('openChange', openChangeHandle);
          document.removeEventListener('click', clickHandler);
          document.removeEventListener('clickEl', clickHandler);
        };
      });
      return story();
    },
  ],
};

const navContent = html`
  <ino-img
    slot="logo"
    src="https://app-uat.james.inma.inovex.io/static/media/james-logo.e7e70758.png"
    width="50"
  ></ino-img>
  <div slot="subtitle">inovex</div>

  <ino-list slot="content">
    <ino-nav-item text="Some Link" activated>
      <ino-icon icon="onboarding"></ino-icon>
    </ino-nav-item>
    <ino-nav-item text="Some other Link">
      <ino-icon icon="calendar"></ino-icon>
    </ino-nav-item>
    <ino-nav-item text="Some other Link">
      <ino-icon icon="employee"></ino-icon>
    </ino-nav-item>
    <ino-list-divider></ino-list-divider>
    <ino-nav-item text="Some settings link">
      <ino-icon icon="settings"></ino-icon>
    </ino-nav-item>
  </ino-list>

  <ino-list slot="footer">
    <ino-nav-item text="My Profile">
      <ino-img
        src="https://picsum.photos/id/1027/250/250.jpg"
        style="border-radius: 50%"
        ratio-width="1"
        ratio-height="1"
      >
      </ino-img>
    </ino-nav-item>
  </ino-list>
`;

const mainContent = html`
  <p>
    At some point someone will replace this block of text with useful words so
    customers can learn more about the products and services you offer!
    Placeholder text is useful when you need to see what a page design looks
    like, but the actual content isn't available. It's like having someone with
    identical measurements check the fit of a dress before trying it on
    yourself.
  </p>
  <p>
    You are currently reading Honest Ipsum, the placeholder text that needs no
    explanation. Determining whether the typeface works or not is only possible
    if there is text for it to be applied to. This is just temporary placeholder
    text; like when a friend saves a spot for you in line, only to be replaced
    by you when you return.
  </p>
  <p>
    A web developer will often use filler text so they can focus on the design
    of the web page. It will be replaced with real content later. It is useful
    for a web developer to use placeholder text so they can easily see what
    different fonts look like on a realistic paragraph.
  </p>
  <p>
    Once the final copy for the web page has been created, it will go here. You
    are currently reading some filler text. A web developer will often use
    filler text so they can focus on the design of the web page. It will be
    replaced with real content later.
  </p>
  <p>
    We aren't quite sure what to put here yet. Determining whether the typeface
    works or not is only possible if there is text for it to be applied to. This
    text is only here to validate the layout. It isn't worth reading. What you
    are reading now is not what you will be reading in this space once this web
    page is completed.
  </p>
  <p>
    We aren't quite sure what to put here yet. Determining whether the typeface
    works or not is only possible if there is text for it to be applied to. This
    text is only here to validate the layout. It isn't worth reading. What you
    are reading now is not what you will be reading in this space once this web
    page is completed.
  </p>
`;

export const Playground = (args) => html`
  <div class="story-nav-drawer__default">
    <ino-nav-drawer
      open=${args.open}
      anchor="${args.anchor}"
      variant="${args.variant}"
      class="customizable-drawer"
    >
      <div slot="header">
        <ino-img
          slot="logo"
          src="https://app-uat.james.inma.inovex.io/static/media/james-logo.e7e70758.png"
          width="50"
        ></ino-img>
        <p>inovex</p>
      </div>
      <ino-list slot="content">
        <ino-nav-item text="Some Link" activated>
          <ino-icon icon="onboarding"></ino-icon>
        </ino-nav-item>
      </ino-list>
      <ino-list slot="footer">
        <ino-nav-item text="My Profile">
          <ino-icon icon="employee"></ino-icon>
        </ino-nav-item>
      </ino-list>

      <main slot="app" class="main-content">
        <ino-button fill="outline" dense class="toggle-nav"
          >Toggle Navigation</ino-button
        >
        <br />
        <br />
        Your App goes here ðŸ¤˜ ${mainContent}
        <br /><br />
      </main>
    </ino-nav-drawer>
  </div>
`;
Playground.args = {
  open: true,
  anchor: 'left',
  variant: 'docked',
};
Playground.argTypes = {
  anchor: {
    control: {
      type: 'select',
    },
    options: ['left', 'right'],
  },
  variant: {
    control: {
      type: 'select',
    },
    options: ['docked', 'dismissible', 'modal'],
  },
};

export const Dismissible = () => {
  return html`
    <div class="story-nav-drawer__dismissible">
      <ino-nav-drawer
        variant="dismissible"
        class="nav-drawer-dismissible"
        anchor="left"
      >
        ${navContent}
        <main slot="app" class="main-content">
          <ino-button fill="outline" dense class="toggle-nav"
            >Toggle Navigation</ino-button
          >
          ${mainContent}
        </main>
      </ino-nav-drawer>
    </div>
  `;
};

export const Modal = () => {
  return html`
    <div class="story-nav-drawer__modal">
      <ino-nav-drawer variant="modal" class="nav-drawer-modal" anchor="left">
        ${navContent}
        <main slot="app" class="main-content">
          <ino-button fill="outline" dense class="toggle-nav"
            >Toggle Navigation</ino-button
          >
          ${mainContent}
        </main>
      </ino-nav-drawer>
    </div>
  `;
};
