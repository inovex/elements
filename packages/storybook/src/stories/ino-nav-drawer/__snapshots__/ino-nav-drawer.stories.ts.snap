// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structure/ino-nav-drawer AnchorRight smoke-test 1`] = `
<div id="root-inner">
  <div class="story-nav-drawer__default">
    <ino-nav-drawer class="customizable-drawer hydrated"
                    open="true"
                    anchor="right"
                    variant="docked"
    >
      <div slot="header">
        <ino-img slot="logo"
                 width="50"
                 src="static/media/src/assets/images/james-logo.png"
                 class="hydrated"
                 style="width: 50px;"
        >
          <div class="ino-img__composer"
               style="padding-top: 100%;"
          >
            <img class="ino-img__image"
                 src="static/media/src/assets/images/james-logo.png"
                 width="50"
            >
            <div class="ino-img__fallback-icon">
              <ino-icon class="ino-icon--color-scheme-primary hydrated">
              </ino-icon>
            </div>
          </div>
        </ino-img>
        <p>
          inovex
        </p>
      </div>
      <ino-list slot="content"
                class="hydrated"
      >
        <ul class="mdc-deprecated-list">
          <ino-nav-item text="Some Link"
                        activated
                        class="hydrated"
          >
            <ino-list-item class="hydrated">
              <li class="mdc-deprecated-list-item mdc-deprecated-list-item--activated mdc-ripple-upgraded"
                  tabindex="-1"
              >
                <span class="mdc-deprecated-list-item__ripple">
                </span>
                <span class="mdc-deprecated-list-item__graphic"
                      role="presentation"
                >
                  <span class="ino-nav-item--leading-slot"
                        slot="leading"
                  >
                    <ino-icon icon="onboarding"
                              class="ino-icon--color-scheme-primary hydrated"
                    >
                    </ino-icon>
                  </span>
                </span>
                <span class="mdc-deprecated-list-item__text">
                  Some Link
                </span>
              </li>
            </ino-list-item>
          </ino-nav-item>
        </ul>
      </ino-list>
      <ino-list slot="footer"
                class="hydrated"
      >
        <ul class="mdc-deprecated-list">
          <ino-nav-item text="My Profile"
                        class="hydrated"
          >
            <ino-list-item class="hydrated">
              <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
                  tabindex="-1"
              >
                <span class="mdc-deprecated-list-item__ripple">
                </span>
                <span class="mdc-deprecated-list-item__graphic"
                      role="presentation"
                >
                  <span class="ino-nav-item--leading-slot"
                        slot="leading"
                  >
                    <ino-icon icon="employee"
                              class="ino-icon--color-scheme-primary hydrated"
                    >
                    </ino-icon>
                  </span>
                </span>
                <span class="mdc-deprecated-list-item__text">
                  My Profile
                </span>
              </li>
            </ino-list-item>
          </ino-nav-item>
        </ul>
      </ino-list>
      <main slot="app"
            class="main-content"
      >
        <ino-button fill="outline"
                    dense
                    class="toggle-nav hydrated"
        >
          <button class="button button__variant--filled button--dense"
                  type="button"
          >
            <span class="button__label">
              Toggle Navigation
            </span>
          </button>
        </ino-button>
        <br>
        <br>
        Your App goes here ðŸ¤˜
        <br>
        <br>
      </main>
    </ino-nav-drawer>
  </div>
</div>
`;

exports[`Structure/ino-nav-drawer Dismissible smoke-test 1`] = `
<div id="root-inner">
  <div class="story-nav-drawer__default">
    <ino-nav-drawer class="customizable-drawer hydrated"
                    open="true"
                    anchor="left"
                    variant="dismissible"
    >
      <div slot="header">
        <ino-img slot="logo"
                 width="50"
                 src="static/media/src/assets/images/james-logo.png"
                 class="hydrated"
                 style="width: 50px;"
        >
          <div class="ino-img__composer"
               style="padding-top: 100%;"
          >
            <img class="ino-img__image"
                 src="static/media/src/assets/images/james-logo.png"
                 width="50"
            >
            <div class="ino-img__fallback-icon">
              <ino-icon class="ino-icon--color-scheme-primary hydrated">
              </ino-icon>
            </div>
          </div>
        </ino-img>
        <p>
          inovex
        </p>
      </div>
      <ino-list slot="content"
                class="hydrated"
      >
        <ul class="mdc-deprecated-list">
          <ino-nav-item text="Some Link"
                        activated
                        class="hydrated"
          >
            <ino-list-item class="hydrated">
              <li class="mdc-deprecated-list-item mdc-deprecated-list-item--activated mdc-ripple-upgraded"
                  tabindex="-1"
              >
                <span class="mdc-deprecated-list-item__ripple">
                </span>
                <span class="mdc-deprecated-list-item__graphic"
                      role="presentation"
                >
                  <span class="ino-nav-item--leading-slot"
                        slot="leading"
                  >
                    <ino-icon icon="onboarding"
                              class="ino-icon--color-scheme-primary hydrated"
                    >
                    </ino-icon>
                  </span>
                </span>
                <span class="mdc-deprecated-list-item__text">
                  Some Link
                </span>
              </li>
            </ino-list-item>
          </ino-nav-item>
        </ul>
      </ino-list>
      <ino-list slot="footer"
                class="hydrated"
      >
        <ul class="mdc-deprecated-list">
          <ino-nav-item text="My Profile"
                        class="hydrated"
          >
            <ino-list-item class="hydrated">
              <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
                  tabindex="-1"
              >
                <span class="mdc-deprecated-list-item__ripple">
                </span>
                <span class="mdc-deprecated-list-item__graphic"
                      role="presentation"
                >
                  <span class="ino-nav-item--leading-slot"
                        slot="leading"
                  >
                    <ino-icon icon="employee"
                              class="ino-icon--color-scheme-primary hydrated"
                    >
                    </ino-icon>
                  </span>
                </span>
                <span class="mdc-deprecated-list-item__text">
                  My Profile
                </span>
              </li>
            </ino-list-item>
          </ino-nav-item>
        </ul>
      </ino-list>
      <main slot="app"
            class="main-content"
      >
        <ino-button fill="outline"
                    dense
                    class="toggle-nav hydrated"
        >
          <button class="button button__variant--filled button--dense"
                  type="button"
          >
            <span class="button__label">
              Toggle Navigation
            </span>
          </button>
        </ino-button>
        <br>
        <br>
        Your App goes here ðŸ¤˜
        <br>
        <br>
      </main>
    </ino-nav-drawer>
  </div>
</div>
`;

exports[`Structure/ino-nav-drawer Modal smoke-test 1`] = `
<div id="root-inner">
  <div class="story-nav-drawer__default">
    <ino-nav-drawer class="customizable-drawer hydrated"
                    open="false"
                    anchor="left"
                    variant="modal"
    >
      <div slot="header">
        <ino-img slot="logo"
                 width="50"
                 src="static/media/src/assets/images/james-logo.png"
                 class="hydrated"
                 style="width: 50px;"
        >
          <div class="ino-img__composer"
               style="padding-top: 100%;"
          >
            <img class="ino-img__image"
                 src="static/media/src/assets/images/james-logo.png"
                 width="50"
            >
            <div class="ino-img__fallback-icon">
              <ino-icon class="ino-icon--color-scheme-primary hydrated">
              </ino-icon>
            </div>
          </div>
        </ino-img>
        <p>
          inovex
        </p>
      </div>
      <ino-list slot="content"
                class="hydrated"
      >
        <ul class="mdc-deprecated-list">
          <ino-nav-item text="Some Link"
                        activated
                        class="hydrated"
          >
            <ino-list-item class="hydrated">
              <li class="mdc-deprecated-list-item mdc-deprecated-list-item--activated mdc-ripple-upgraded"
                  tabindex="-1"
              >
                <span class="mdc-deprecated-list-item__ripple">
                </span>
                <span class="mdc-deprecated-list-item__graphic"
                      role="presentation"
                >
                  <span class="ino-nav-item--leading-slot"
                        slot="leading"
                  >
                    <ino-icon icon="onboarding"
                              class="ino-icon--color-scheme-primary hydrated"
                    >
                    </ino-icon>
                  </span>
                </span>
                <span class="mdc-deprecated-list-item__text">
                  Some Link
                </span>
              </li>
            </ino-list-item>
          </ino-nav-item>
        </ul>
      </ino-list>
      <ino-list slot="footer"
                class="hydrated"
      >
        <ul class="mdc-deprecated-list">
          <ino-nav-item text="My Profile"
                        class="hydrated"
          >
            <ino-list-item class="hydrated">
              <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
                  tabindex="-1"
              >
                <span class="mdc-deprecated-list-item__ripple">
                </span>
                <span class="mdc-deprecated-list-item__graphic"
                      role="presentation"
                >
                  <span class="ino-nav-item--leading-slot"
                        slot="leading"
                  >
                    <ino-icon icon="employee"
                              class="ino-icon--color-scheme-primary hydrated"
                    >
                    </ino-icon>
                  </span>
                </span>
                <span class="mdc-deprecated-list-item__text">
                  My Profile
                </span>
              </li>
            </ino-list-item>
          </ino-nav-item>
        </ul>
      </ino-list>
      <main slot="app"
            class="main-content"
      >
        <ino-button fill="outline"
                    dense
                    class="toggle-nav hydrated"
        >
          <button class="button button__variant--filled button--dense"
                  type="button"
          >
            <span class="button__label">
              Toggle Navigation
            </span>
          </button>
        </ino-button>
        <br>
        <br>
        Your App goes here ðŸ¤˜
        <br>
        <br>
      </main>
    </ino-nav-drawer>
  </div>
</div>
`;

exports[`Structure/ino-nav-drawer Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-nav-drawer__default">
    <ino-nav-drawer class="customizable-drawer"
                    open="true"
                    anchor="left"
                    variant="docked"
    >
      <div slot="header">
        <ino-img slot="logo"
                 width="50"
                 src="static/media/src/assets/images/james-logo.png"
                 class="hydrated"
                 style="width: 50px;"
        >
          <div class="ino-img__composer"
               style="padding-top: 100%;"
          >
            <img class="ino-img__image"
                 src="static/media/src/assets/images/james-logo.png"
                 width="50"
            >
            <div class="ino-img__fallback-icon">
              <ino-icon class="hydrated">
              </ino-icon>
            </div>
          </div>
        </ino-img>
        <p>
          inovex
        </p>
      </div>
      <ino-list slot="content">
        <ul class="mdc-deprecated-list">
          <ino-nav-item text="Some Link"
                        activated
          >
            <ino-list-item>
              <span class="ino-nav-item--leading-slot"
                    slot="leading"
              >
                <ino-icon icon="onboarding"
                          class="hydrated"
                >
                </ino-icon>
              </span>
            </ino-list-item>
          </ino-nav-item>
        </ul>
      </ino-list>
      <ino-list slot="footer">
        <ul class="mdc-deprecated-list">
          <ino-nav-item text="My Profile">
            <ino-list-item>
              <span class="ino-nav-item--leading-slot"
                    slot="leading"
              >
                <ino-icon icon="employee"
                          class="hydrated"
                >
                </ino-icon>
              </span>
            </ino-list-item>
          </ino-nav-item>
        </ul>
      </ino-list>
      <main slot="app"
            class="main-content"
      >
        <ino-button fill="outline"
                    dense
                    class="toggle-nav hydrated"
        >
          <button class="button button__variant--filled button--dense"
                  type="button"
          >
            <span class="button__label">
              Toggle Navigation
            </span>
          </button>
        </ino-button>
        <br>
        <br>
        Your App goes here ðŸ¤˜
        <br>
        <br>
      </main>
    </ino-nav-drawer>
  </div>
</div>
`;
